frappe/social/doctype/energy_point_log/energy_point_log.py

```diff
@@ -171,7 +171,13 @@ def create_energy_points_log(ref_doctype, ref_name, doc, apply_only_once=False):
 		ref_name, doc.rule, None if apply_only_once else doc.user)
 
 	if log_exists:
-		return
+		filters = frappe._dict({
+			'rule': doc.rule,
+			'reference_doctype': ref_doctype,
+			'reference_name': ref_name,
+			'reverted': 0
+		})
+		return frappe.get_doc('Energy Point Log', filters=filters)
```

Comment:
`frappe.db.exists` returns `name` if document exists. you can use `log_exists` directly :p
```suggestion
		return frappe.get_doc('Energy Point Log', name=log_exists)
```